cognite/neat/utils/cdf_loaders/_base.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from typing import Generic, TypeVar
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes._base import (
+    T_CogniteResourceList,
+    T_WritableCogniteResource,
+    T_WriteClass,
+    WriteableCogniteResourceList,
+)
+from cognite.client.data_classes.data_modeling import (
+    DataModelingId,
+)
+from cognite.client.data_classes.data_modeling.ids import (
+    InstanceId,
+    VersionedDataModelingId,
+)
+from cognite.client.utils.useful_types import SequenceNotStr
+
+from .data_classes import RawTableID
+
+T_ID = TypeVar("T_ID", bound=str | int | DataModelingId | InstanceId | VersionedDataModelingId | RawTableID)
+T_WritableCogniteResourceList = TypeVar("T_WritableCogniteResourceList", bound=WriteableCogniteResourceList)
+
+
+class ResourceLoader(
+    ABC,
+    Generic[T_ID, T_WriteClass, T_WritableCogniteResource, T_CogniteResourceList, T_WritableCogniteResourceList],
+):
+    resource_name: str
+
+    def __init__(self, client: CogniteClient) -> None:
+        self.client = client
+
+    @classmethod
+    @abstractmethod
+    def get_id(cls, item: T_WriteClass | T_WritableCogniteResource) -> T_ID:
```

Comment:
suggestion maybe rename this to `as_id` ? wandering do you need two separate methods one for listable, one for single items? can there be a single `as_id` method?

Reply:
even better call this `as_external_ids` to comply with sdk

Reply:
Hmm, I want to distinguish this from `.as_id()`, `.as_ids()` which are on most Cognite Resource objects.